plugins/transport/socket/main.go

```diff
@@ -80,17 +81,48 @@ func (s *Socket) Run(ctx context.Context, w transport.WriteFn, done chan bool) {
 	pc, err := net.ListenUnixgram("unixgram", &laddr)
 	if err != nil {
 		s.logger.Errorf(err, "failed to bind unix socket %s", laddr.Name)
-		return
+		return nil
 	}
 	// create socket file if it does not exist
 	skt, err := pc.File()
 	if err != nil {
 		s.logger.Errorf(err, "failed to retrieve file handle for %s", laddr.Name)
-		return
+		return nil
 	}
 	skt.Close()
 
 	s.logger.Infof("socket listening on %s", laddr.Name)
+
+	return pc
+}
+
+func (s *Socket) initUdpSocket() *net.UDPConn {
+	addr, err := net.ResolveUDPAddr("udp", s.conf.Url)
+	pc, err := net.ListenUDP("udp", addr)
+	if err != nil {
+		s.logger.Errorf(err, "failed to bind unix socket to url: %s", s.conf.Url)
+		return nil
+	}
+
+	s.logger.Infof("socket listening on %s", s.conf.Url)
+
+	return pc
+}
+
+// Run implements type Transport
+func (s *Socket) Run(ctx context.Context, w transport.WriteFn, done chan bool) {
+	var pc net.Conn
+	if s.conf.Type == "unix" {
+		pc = s.initUnixSocket()
+	} else if s.conf.Type == "udp" {
+		pc = s.initUdpSocket()
+	} else {
+		s.conf.Type = "unix"
```

Comment:
What is the purpose of this assignment?

Reply:
This was a "last minute" attempt to fix the unit tests, which don't set the s.conf.Type to anything before our meeting yesterday. In order for this to work, that assignment would need to be before the first "if". Sorry for that